src/main/java/net/fabricmc/nameproposal/NameProposalConfig.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2023 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.nameproposal;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import net.fabricmc.nameproposal.field.nameprovider.FieldNameProvider;
+import net.fabricmc.nameproposal.field.nameprovider.FieldNameProviders;
+import net.fabricmc.nameproposal.registry.Codecs;
+
+public record NameProposalConfig(FieldNameProvider fieldNameProvider, int version) {
+	public static final String FILE_NAME = "name_proposal.json";
```

Comment:
Remove this, shouldnt be hard coded. See the other comment about passing this to enigma.